src/foam/crypto/sign/SigningDAO.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.crypto.sign;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import org.bouncycastle.util.encoders.Hex;
+
+import java.security.PrivateKey;
+
+public class SigningDAO
+    extends ProxyDAO
+{
+  protected PrivateKey key_;
+  protected String algorithm_;
+  protected DAO signatureDAO_;
+
+  // TODO: come up with a better way to set the algorithm and key than this
+  public SigningDAO(X x, DAO delegate, String algorithm, PrivateKey key) {
```

Comment:
Can we get a default constructor which a reasonable default algorithm already set?

Reply:
Yeah, just added one